tests/src/Unit/ParserTest.php

```diff
@@ -231,6 +233,21 @@ public function testVersionSplitting()
             ['@gulp-sourcemaps/identity-map', '1.X'],
             Parser::splitVersionString('@gulp-sourcemaps/identity-map@1.X')
         );
+
+        static::assertSame(
+            ['@foo/bar', 'git+ssh://user@host:1234/foo/bar#semver:^1.2.3'],
+            Parser::splitVersionString('@foo/bar@git+ssh://user@host:1234/foo/bar#semver:^1.2.3')
+        );
+
+        static::assertSame(
+            ['@foo/bar', 'git://user@host/foo/bar.git#v1.2.3'],
+            Parser::splitVersionString('@foo/bar@git://user@host/foo/bar.git#v1.2.3')
+        );
+
+        static::assertSame(
+            ['@foo/bar', 'file:vendor/foo/bar'],
+            Parser::splitVersionString('@foo/bar@file:vendor/foo/bar')
+        );
```

Comment:
Could you please add an example with a non-namespaced package for completeness?

If you prefer, you can also restructure this to use a PHPUnit dataprovider function.